aten/src/ATen/native/mkldnn/xpu/qconv.cpp

```diff
@@ -0,0 +1,124 @@
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/mkldnn/xpu/detail/oneDNN.h>
+#include <c10/core/MemoryFormat.h>
+#include <torch/library.h>
+
+#include <iostream>
+
+using namespace at::native::onednn;
+namespace at::native::xpu {
+
+at::Tensor qconv_prepack_xpu(
+    at::Tensor weight,
+    at::Tensor weight_scales,
+    double input_scale,
+    int64_t input_zero_point,
+    torch::List<int64_t> stride,
+    torch::List<int64_t> padding,
+    torch::List<int64_t> dilation,
+    int64_t groups,
+    std::optional<torch::List<int64_t>> input_shape) {
+  // XPU has no prepack at present
+  return weight;
+}
+
+class QConvoneDNNXPU final {
+ public:
+  static at::Tensor run_pointwise(
+      at::Tensor act,
+      double act_scale,
+      int64_t act_zero_point,
+      at::Tensor weight,
+      at::Tensor weight_scales,
+      at::Tensor weight_zero_points,
+      c10::optional<at::Tensor> bias,
+      torch::List<int64_t> stride,
+      torch::List<int64_t> padding,
+      torch::List<int64_t> dilation,
+      int64_t groups,
+      double inv_output_scale,
+      int64_t output_zero_point,
+      c10::optional<c10::ScalarType> output_dtype,
+      c10::string_view attr,
+      torch::List<c10::optional<at::Scalar>> scalars,
+      c10::optional<c10::string_view> algorithm) {
+    if (act.dim() == 3 || act.dim() == 5) {
+      TORCH_CHECK(
+          attr == "none",
+          "quantized pointwise conv",
+          act.dim() - 2,
+          "d doesn't support unary_post_op fusion. Got unary_post_op:",
+          attr,
+          ".");
+    } else {
+      TORCH_CHECK(
+          attr == "none" || attr == "relu" || attr == "hardtanh" ||
+              attr == "hardswish" || attr == "swish",
+          "We support quantized convolution without post-op(attr==None) and combinations for Quantized Conv + ReLU, Hardtanh, and Hardswish are supported. However, encountered unsupported post operation:",
```

Comment:
Do you think users understand the exact meaning of `post-op(attr==None)` ?

Reply:
Has changed it to ```We support quantized convolution without any post-ops or combinations for Quantized Conv + ReLU, Hardtanh, and Hardswish are supported. However, encountered unsupported post operation```